README.md

```diff
@@ -1,41 +1,61 @@
 # Charmil
 
+> TODO Logo
+
 Framework for building command line plugins on top of Golang Cobra Library that allows developers to install and use other (even non golang CLIs) as extensions.
 Charmil will let you control your cobra based CLIs and extend them without embeeding any plugin code into your CLI.
 
 > NOTE: Under construction
 
+## Working
+
+![working](mockups/architecture2.png)
+
 ## Features
 
 - Install any CLI as part of your host CLI (even CLI that you do not control)
 - Build and deploy registry with different versions of CLI's to install that can be hosted in Github Pages or as yaml files
-- Provide set of the cobra commands that will let users to install and manage all available extensions
-
+- Provide set of the cobra commands that will let users to install, update, remove or manage all available extensions
 
-## Commands that will be added to your CLI by Charmil
-
-`$ yourCLi extensions list`
-
-`$ yourCLi extensions installed` # list installed ones
+## Getting started
 
-`$ yourCLi extensions install <name>@<version>`
+### Adding library to your existing CLI
 
-`$ yourCLi extensions remove <name>@<version>`
+Include Charmil in your CLI by doing `$ go get -u github.com/aerogear/charmil`
 
+#### Commands that will be added to your CLI by Charmil
 
-## Getting started
+1. List all the available extensions
+   > `$ yourCLi extensions list`
+2. List all the installed extensions
+   > `$ yourCLi extensions installed`
```

Comment:
Those will not be the commands. Please review commands that kew provides:
https://github.com/kubernetes-sigs/krew

Idealy we could keep the same pattern.